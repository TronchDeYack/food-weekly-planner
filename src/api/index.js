const recipeFake = {
  id: 'SPAGHETTI-ID',
  title: 'Spaghetti bolognaise',
  minutes: 20,
  caloriesLevel: 'medium',
  ingredients: [
    {
      id: 'PATE-ID',
      label: 'Pâtes',
      quantity: 200,
      unit: 'GR',
      type: 'STARCHY-FOOD',
    },
    {
      id: 'SAUCE-TOMATE-ID',
      label: 'Sauce Tomate',
      quantity: 200,
      unit: 'GR',
      type: 'CONDIMENT',
    },
    {
      id: 'STEAK-ID',
      label: 'Steak haché',
      quantity: 100,
      unit: 'GR',
      type: 'MEAT',
    },
    {
      id: 'OIL-ID',
      label: 'Huile d\'olives',
      quantity: 3,
      unit: 'CAC',
      type: 'CONDIMENT',
    },
  ],
};

const recipesFake = [
  recipeFake,
  {
    id: 'SOUPE-ID',
    title: 'Soupe à la tomate',
    minutes: 30,
    caloriesLevel: 'low',
    ingredients: [
      {
        id: 'TOMATE-ID',
        label: 'Tomates',
        quantity: 4,
        unit: 'UNIT',
        type: 'VEGETABLE',
      },
      {
        id: 'SALT-ID',
        label: 'Sel',
        quantity: 2,
        unit: 'CAC',
        type: 'CONDIMENT',
      },
    ],
  },
  {
    id: 'WELSH-ID',
    title: 'Welsh',
    minutes: 45,
    caloriesLevel: 'high',
    ingredients: [
      {
        id: 'CHEDDAR-ID',
        label: 'Cheddar',
        quantity: 400,
        unit: 'GR',
        type: 'DAIRY-PRODUCT',
      },
      {
        id: 'BEER-ID',
        label: 'Bière brune',
        quantity: 25,
        unit: 'CL',
        type: 'OTHER',
      },
      {
        id: 'HAM-ID',
        label: 'Jambon',
        quantity: 100,
        unit: 'GR',
        type: 'MEAT',
      },
      {
        id: 'BREAD-ID',
        label: 'Pain',
        quantity: 4,
        unit: 'SLICE',
        type: 'STARCHY-FOOD',
      },
    ],
  },
];

export function getRecipe(id) {
  console.log(`get recipe ${id}`);
  return Promise.resolve(recipeFake);
}

export function getRecipes() {
  return Promise.resolve(recipesFake);
}

export function getRecipesOfTheWeek() {
  return Promise.resolve([recipesFake[1], recipesFake[2]]);
}

export function getShoppingList() {
  return Promise.resolve([
    {
      id: 'BREAD-ID',
      label: 'Pain',
      quantity: 1,
      unit: 'unit',
      type: 'STARCHY-FOOD',
      bought: false,
    },
    {
      id: 'CHEDDAR-ID',
      label: 'Cheddar',
      quantity: 400,
      unit: 'GR',
      type: 'DAIRY-PRODUCT',
      bought: true,
    },
    {
      id: 'MILK-ID',
      label: 'Lait',
      quantity: 1,
      unit: 'L',
      type: 'DAIRY-PRODUCT',
      bought: false,
    },
    {
      id: 'BEER-ID',
      label: 'Bière brune',
      quantity: 25,
      unit: 'CL',
      type: 'OTHER',
      bought: false,
    },
    {
      id: 'HAM-ID',
      label: 'Jambon',
      quantity: 100,
      unit: 'GR',
      type: 'MEAT',
      bought: true,
    },
    {
      id: 'CHICKEN-ID',
      label: 'Piou piou',
      quantity: 200,
      unit: 'GR',
      type: 'MEAT',
      bought: false,
    },
  ]);
}

export function getIngredients() {
  return Promise.resolve([
    {
      id: 'PATE-ID',
      label: 'Pâtes',
      quantity: 200,
      unit: 'GR',
      type: 'STARCHY-FOOD',
    },
    {
      id: 'SAUCE-TOMATE-ID',
      label: 'Sauce Tomate',
      quantity: 200,
      unit: 'GR',
      type: 'CONDIMENT',
    },
    {
      id: 'STEAK-ID',
      label: 'Steak haché',
      quantity: 100,
      unit: 'GR',
      type: 'MEAT',
    },
    {
      id: 'OIL-ID',
      label: 'Huile d\'olives',
      quantity: 3,
      unit: 'CAC',
      type: 'CONDIMENT',
    },
    {
      id: 'TOMATE-ID',
      label: 'Tomates',
      quantity: 4,
      unit: 'UNIT',
      type: 'VEGETABLE',
    },
    {
      id: 'SALT-ID',
      label: 'Sel',
      quantity: 2,
      unit: 'CAC',
      type: 'CONDIMENT',
    },
    {
      id: 'CHEDDAR-ID',
      label: 'Cheddar',
      quantity: 400,
      unit: 'GR',
      type: 'DAIRY-PRODUCT',
    },
    {
      id: 'BEER-ID',
      label: 'Bière brune',
      quantity: 25,
      unit: 'CL',
      type: 'OTHER',
    },
    {
      id: 'HAM-ID',
      label: 'Jambon',
      quantity: 100,
      unit: 'GR',
      type: 'MEAT',
    },
    {
      id: 'BREAD-ID',
      label: 'Pain',
      quantity: 4,
      unit: 'SLICE',
      type: 'STARCHY-FOOD',
    },
  ]);
}
