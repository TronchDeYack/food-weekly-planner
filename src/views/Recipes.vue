<template>
  <v-container>
    <h2 class="mt-2">Recipes</h2>
    <v-row>
      <v-col>
        <MenuCard title="Spaghetti bolognaise" :minutes=20
          calories-indicator="medium" :has-egg="false" :has-fish="false"
          :has-meat="true" :is-vegetarian="false" />
      </v-col>
    </v-row>
    <v-btn
      absolute
      dark
      fab
      bottom
      right
      color="secondary"
      class="btn-over-navbar"
      to="/add-recipe"
    >
      <v-icon>mdi-plus</v-icon>
    </v-btn>
  </v-container>
</template>


<script>
import MenuCard from '@/components/MenuCard.vue';

export default {
  name: 'recipes',
  components: {
    MenuCard,
  },
};
</script>
