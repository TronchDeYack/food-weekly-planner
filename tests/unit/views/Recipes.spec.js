import { shallowMount } from '@vue/test-utils';

import Recipes from '@/views/Recipes.vue';
import { getRecipes } from '@/api';

jest.mock('@/api');

describe('Recipes', () => {
  const recipes = [
    {
      id: 'SOUPE-ID',
      title: 'Soupe à la tomate',
      minutes: 30,
      caloriesLevel: 'low',
      ingredients: [
        {
          id: 'TOMATE-ID',
          label: 'Tomates',
          quantity: 4,
          unit: 'UNIT',
          type: 'VEGETABLE',
        },
        {
          id: 'SALT-ID',
          label: 'Sel',
          quantity: 2,
          unit: 'CAC',
          type: 'CONDIMENT',
        },
      ],
    },
    {
      id: 'WELSH-ID',
      title: 'Welsh',
      minutes: 45,
      caloriesLevel: 'high',
      ingredients: [
        {
          id: 'CHEDDAR-ID',
          label: 'Cheddar',
          quantity: 400,
          unit: 'GR',
          type: 'DAIRY-PRODUCT',
        },
        {
          id: 'BEER-ID',
          label: 'Bière brune',
          quantity: 25,
          unit: 'CL',
          type: 'OTHER',
        },
        {
          id: 'HAM-ID',
          label: 'Jambon',
          quantity: 100,
          unit: 'GR',
          type: 'MEAT',
        },
        {
          id: 'BREAD-ID',
          label: 'Pain',
          quantity: 4,
          unit: 'SLICE',
          type: 'STARCHY-FOOD',
        },
      ],
    },
  ];

  let wrapper;

  beforeEach(() => {
    getRecipes.mockResolvedValue(recipes);
    wrapper = shallowMount(Recipes);
  });

  describe('render', () => {
    it('Should render.', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('Should display the menu card with the data of the given recipes.', () => {
      recipes.forEach((recipe, index) => {
        expect(wrapper.findAll('menucard-stub').at(index).props('id')).toEqual(recipe.id);
        expect(wrapper.findAll('menucard-stub').at(index).props('title')).toEqual(recipe.title);
        expect(wrapper.findAll('menucard-stub').at(index).props('caloriesLevel')).toEqual(recipe.caloriesLevel);
        expect(wrapper.findAll('menucard-stub').at(index).props('ingredients')).toEqual(recipe.ingredients);
        expect(wrapper.findAll('menucard-stub').at(index).props('minutes')).toEqual(recipe.minutes);
      });
    });
  });

  describe('State', () => {
    it('Should initialize "recipes" to an empty array.', () => {
      const cmp = shallowMount(Recipes);
      expect(cmp.vm.$data.recipes).toEqual([]);
    });
  });

  describe('mounted', () => {
    it('Should get recipes data from API and store them in "recipes" state.', () => {
      expect(wrapper.vm.$data.recipes).toEqual(recipes);
    });
  });
});
