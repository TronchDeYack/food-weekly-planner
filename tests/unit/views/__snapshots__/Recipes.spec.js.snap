// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipes Should render. 1`] = `
<v-container-stub tag="div">
  <h2 class="mt-2">Recipes</h2>
  <v-row-stub tag="div">
    <v-col-stub tag="div">
      <menucard-stub id="RECIPE-1" title="Spaghetti bolognaise" minutes="20" caloriesindicator="medium" hasmeat="true"></menucard-stub>
    </v-col-stub>
  </v-row-stub>
  <v-btn-stub color="secondary" dark="true" tag="button" activeclass="" to="/add-recipe" absolute="true" bottom="true" right="true" fab="true" type="button" class="btn-over-navbar">
    <v-icon-stub>mdi-plus</v-icon-stub>
  </v-btn-stub>
</v-container-stub>
`;
